src/libraries/Microsoft.PowerFx.Core/Functions/TexlFunction.cs

```diff
@@ -1011,76 +1025,115 @@ protected bool CheckType(TexlNode node, DType nodeType, DType expectedType, IErr
+        // Coercion is supported and nodeToCoercedTypeMap is updated.
+        //
+        // Many functions return the same type and column name (in the days before ConsistentOneColumnTableResult) as the first argument,
+        // pass context (to get at features) and an out parameter for the type to use as the result of such a function.
+        // If not needed, use one of the overloads without context and the out parameter.
+        // Date/DateTime/Time are special and will not snap to expectedType but will retain the subtype.
+        private bool CheckColumnType(DType type, TexlNode arg, DType expectedType, IErrorContainer errors, ErrorResourceKey errKey, ref Dictionary<TexlNode, DType> nodeToCoercedTypeMap, CheckTypesContext context, out DType returnType)
         {
             Contracts.Assert(type.IsValid);
             Contracts.AssertValue(arg);
             Contracts.Assert(expectedType.IsValid);
-            Contracts.AssertValue(errors);
-
+            Contracts.AssertValue(errors);
+
+            returnType = type;
+
             IEnumerable<TypedName> columns;
             if (!type.IsTable || (columns = type.GetNames(DPath.Root)).Count() != 1)
             {
                 errors.EnsureError(DocumentErrorSeverity.Severe, arg, TexlStrings.ErrInvalidSchemaNeedCol);
                 return false;
             }
-            else
+
+            var column = columns.Single();
+            var colType = column.Type;
+
+            if (!expectedType.Accepts(colType))
             {
-                var column = columns.Single();
-                if (!expectedType.Accepts(column.Type))
+                if (SupportsParamCoercion && column.Type.CoercesTo(expectedType))
+                {
+                    returnType = DType.CreateTable(new TypedName(expectedType, column.Name));
+                    CollectionUtils.Add(ref nodeToCoercedTypeMap, arg, returnType);
+                }
+                else
                 {
-                    if (SupportsParamCoercion && column.Type.CoercesTo(expectedType))
-                    {
-                        expectedType = DType.CreateTable(new TypedName(expectedType, column.Name));
-                        CollectionUtils.Add(ref nodeToCoercedTypeMap, arg, expectedType);
-                    }
-                    else
-                    {
-                        errors.EnsureError(DocumentErrorSeverity.Severe, arg, errKey, column.Name.Value);
-                        return false;
-                    }
+                    errors.EnsureError(DocumentErrorSeverity.Severe, arg, errKey, column.Name.Value);
+                    return false;
                 }
             }
-
+
+            if (context != null && context.Features.ConsistentOneColumnTableResult)
+            {
+                // Note that DateTime retains the subtype and does not snap to expectedType

```

Comment:
>Note that DateTime retains the subtype and does not snap to expectedType [](http://example.com/codeflow?start=19&length=72)

What's an example of this special case?

Reply:
Added this comment:
`// For example, DateAdd([Date(2000,1,1)],3) should return *[Value:D] and not *[Value:d]`